### ISSUE TITLE
Prim parenting issues since maya-usd 0.23

### ISSUE THREAD

**Issue by jufrantz** [Autodesk/maya-usd#3237]
**Describe the bug**

We encounter issues while parenting multiple prims with USD references. It seems to be a regression between maya-usd 0.22 and 0.23.
- Maya outliner is not refreshed after parenting action.
- Only the first selected prim is really re-parented. 
- Undo stack is broken after the action.

Also, very similar problems occur with group command / `Ctlr+G`. I can file another issue if needed.

**Steps to reproduce**

- In attached [maya-usd-23_parenting_issues.zip](https://github.com/Autodesk/maya-usd/files/12094630/maya-usd-23_parenting_issues.zip), open maya scene `scene.ma`. 
    - `scene.ma` contains a `mayaUsdProxyShape` created from `scene.usd` layer. 
    - `scene.usd` defines a simple hierarchy with multiple references to `asset.usd`.
   
- Select ufe items in this order `/World/Group1/Asset1` and `/World/Group1/Asset2` (both referencing `asset.usd`) and `/World/Group2` (the wanted parent).
- Click on menu `Edit/Parent` or use shortcut `P`
- Notice that 
  - Maya outliner is out of date, you need to collapse/expand `Group1` and `World` to see the new parenting.
  - `Asset2` was not parented, parent action was only applied to first selected prim `Asset1`.
  - Next `undo` will issue this error
```
// Error: line 1: Undo was not successful.  Damaged undo/redo information has been flushed.
```

- The bug can also be reproduced executing following script
```python
import maya.cmds
maya.cmds.parent(
    "|usdProxy|usdProxyShape,/World/Group1/Asset1", 
    "|usdProxy|usdProxyShape,/World/Group1/Asset2",
    "|usdProxy|usdProxyShape,/World/Group2"
)

```

**Specs (if applicable):**
 - Linux CentOS 7.8
 - gcc 9.3.1
 - Maya 2023.3
 - Maya USD 0.23.0
 - Pixar USD 22.11

**Comment by jufrantz** [comment]
Hello @santosd.

Good to know. Thank you for the information and for testing this case. We'll try 0.24 soon. 

A bit unrelated, but to anticipate the way we would deploy next version : maya-usd 0.24 dropped support for Maya 2020. Is it the case even if we build maya-usd from sources. Or does it only mean that the prebuilt release packages are no longer targeting 2020 ? Sorry if I missed this information elsewhere.

Thank you once again,
Julien

**Comment by seando-adsk** [comment]
You should be able to build your own 0.24 targeting Maya 2020 as support for it was removed after we released 0.24. Future releases will require at least Maya 2022.

Sean

**Comment by santosd** [comment]
I will go ahead and close this out since the issue is no longer in 0.24.0, but if you have any further questions on this feel free to reply and reopen or if a new issue comes up feel free to log it.
